packages/snaps-execution-environments/src/common/BaseSnapExecutor.ava.test.ts

```diff
@@ -0,0 +1,223 @@
+    params: [
+      FAKE_SNAP_NAME,
+      ON_RPC_REQUEST,
+      FAKE_ORIGIN,
+      { jsonrpc: '2.0', method: '', params: [] },
+    ],
+  });
+
+  expect.like(await executor.readCommand(), {
+    jsonrpc: '2.0',
+    id: 2,
+    result: {
+      hasMethods: {
+        ethereum: {
+          request: true,
+          on: true,
+          removeListener: true,
+          rpcEngine: false,
+        },
+        snap: {
+          request: true,
+          on: false,
+          removeListener: false,
+          rpcEngine: false,
+          requestType: 'function',
+        },
+      },
+    },
+    errors: undefined,
+  });
+});
+
+const testSubjects = ['ethereum', 'snap'];
+
+testSubjects.forEach((endowment) => {
+  test(`hardening protects ${endowment} endowment`, async (expect) => {
+    const CODE = `
+      module.exports.onRpcRequest = () => {
+        let result = 'ENDOWMENT_SECURED';
+        let errors = [];
+
+        try {
+          errors = (${testEndowmentHardening})(${endowment}, () => ${endowment});
+        } catch (error) {
+          return error.message;
+        }
+
+        try {
+          if (${endowment}.__flag) {
+            result = 'ENDOWMENT_NOT_SECURED';
+          }
+        } catch (error) {
+          return error.message;
+        }
+
+        try {
+          if (${endowment}.__proto__ && ${endowment}.__proto__.__flag) {
+            result = 'ENDOWMENT_NOT_SECURED';
+          }
+        } catch (error) {
+          return error.message;
+        }
+
+        try {
+          if (${endowment}.prototype && ${endowment}.prototype.__flag) {
+            result = 'ENDOWMENT_NOT_SECURED';
+          }
+        } catch (error) {
+          return error.message;
+        }
+
+        try {
+          const objectProto = Object.getPrototypeOf(${endowment});
+          if (objectProto.__flag) {
+            result = 'ENDOWMENT_NOT_SECURED';
+          }
+        } catch (error) {
+          return error.message;
+        }
+
+        return {
+          result: result,
+          errors: errors,
+        };
+      };
+    `;
+    const executor = new TestSnapExecutor();
+
+    await executor.executeSnap(1, FAKE_SNAP_NAME, CODE, ['ethereum']);
+
+    expect.like(await executor.readCommand(), {
+      jsonrpc: '2.0',
+      id: 1,
+      result: 'OK',
+    });
+
+    await executor.writeCommand({
+      jsonrpc: '2.0',
+      id: 2,
+      method: 'snapRpc',
+      params: [
+        FAKE_SNAP_NAME,
+        ON_RPC_REQUEST,
+        FAKE_ORIGIN,
+        { jsonrpc: '2.0', method: '', params: [] },
+      ],
+    });
+
+    const hardeningTestResult = await executor.readCommand();
+    // @ts-expect-error Result should be on the returned object.
+    const hardeningTestErrors = hardeningTestResult.result.errors;
+
+    expect.like(hardeningTestResult, {
+      jsonrpc: '2.0',
+      id: 2,
+      result: {
+        result: 'ENDOWMENT_SECURED',
+      },
+    });
+
+    expect.true(hardeningTestErrors.length > 0);
```

Comment:
it'd be nice to make this assertion inform the reader what the errors are. or at least comment on that.

Reply:
I added a comment explaining why and what's expected, in short. And also added some assertion with the actual error in a way that will work with our current setup with AVA and hardening test function.